governator-api/src/main/java/com/netflix/governator/event/ApplicationEventDispatcher.java

```diff
@@ -0,0 +1,19 @@
+package com.netflix.governator.event;
+
+import java.lang.reflect.Method;
+
+/**
+ * Interface to publishing {@link ApplicationEvent}s as well as programmatically registering
+ * {@link ApplicationEventListener}s.
+ */
+public interface ApplicationEventDispatcher {
+
+    <T extends ApplicationEvent> void registerListener(Class<T> eventType, ApplicationEventListener<T> eventListener);
+
+    void registerListener(ApplicationEventListener<? extends ApplicationEvent> eventListener);
+
+    void registerListener(Object instance, Method method, Class<? extends ApplicationEvent> acceptedType);
```

Comment:
I was exposing this on the off chance someone wanted to register listeners without having any dependencies on governator api's in their service logic.